java/core/src/main/java/sleeper/core/record/serialiser/RecordJSONSerDe.java

```diff
@@ -111,10 +111,8 @@ public Record deserialize(JsonElement jsonElement, java.lang.reflect.Type typeOf
     }
 
     private static void addFieldToJsonObject(Field field, Object fieldValue, JsonObject json) {
-        if (field.getType() instanceof IntType) {
-            json.addProperty(field.getName(), (int) fieldValue);
-        } else if (field.getType() instanceof LongType) {
-            json.addProperty(field.getName(), (long) fieldValue);
+        if (field.getType() instanceof IntType || field.getType() instanceof LongType) {
+            json.addProperty(field.getName(), (Number) fieldValue);
```

Comment:
Casting to Number is less precise and will not throw an exception if eg. BigInteger is used. We can use Integer and Long to avoid unboxing/reboxing.